src/gc-pages.c

```diff
@@ -0,0 +1,209 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+
+#include "gc.h"
+#ifndef _OS_WINDOWS_
+#  include <sys/resource.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// A region is contiguous storage for up to REGION_PG_COUNT naturally aligned GC_PAGE_SZ pages
+// It uses a very naive allocator (see jl_gc_alloc_page & jl_gc_free_page)
+#if defined(_P64) && !defined(_COMPILER_MICROSOFT_)
+#define DEFAULT_REGION_PG_COUNT (16 * 8 * 4096) // 8 GB
+#else
+#define DEFAULT_REGION_PG_COUNT (8 * 4096) // 512 MB
+#endif
+#define MIN_REGION_PG_COUNT 64 // 1 MB
+
+static int region_pg_cnt = DEFAULT_REGION_PG_COUNT;
+static jl_mutex_t pagealloc_lock;
+static size_t current_pg_count = 0;
+
+void jl_gc_init_page(void)
+{
+#ifndef _OS_WINDOWS_
+    struct rlimit rl;
+    if (getrlimit(RLIMIT_AS, &rl) == 0) {
```

Comment:
Hmm, which one? [This](https://github.com/JuliaLang/libuv/blob/a1d9166a440e4a0664c0e6de6ebe25350de56a42/src/unix/thread.c#L78) is the only use of `getrlimit` I found. The code here is just an optimization and is likely irrelevant for windows anyway (there's unlikely a windows cluster with a system manager setting a address space limit for each user....)